src/RuleSet/DeclarationBlock.php

```diff
@@ -585,7 +586,7 @@ public function createShorthandProperties(array $aProperties, $sShorthand)
                 $this->removeRule($sProperty);
             }
         }
-        if (count($aNewValues)) {
+        if ($oRule instanceof Rule && count($aNewValues)) {
```

Comment:
> Wouldn’t a `null` check suffice instead of the `instanceof`?

Technically, it probably would. For the sake of communicating to the reader, I usually prefer the `instanceof` to reduce the cognitive load required to understand the resulting type.

Reply:
> [`instanceof` is quite slow](https://stackoverflow.com/questions/40179278/why-does-using-instanceof-calls-take-so-much-time), whereas [`count` is atomic](https://stackoverflow.com/questions/5835241/is-phps-count-function-o1-or-on-for-arrays). So it would be much more optimal to have `&&`'s parameters the other way round.

Actually not that much more optimal, on reflection, since the `instanceof` test still needs execute whenever the array is nonempty (and the object is nonnull).  Swapping the parameter order only saves the `instanceof` test when the object is `null`, when I would expect it to be quite quick.